doc/tests.rst

```diff
@@ -448,8 +449,151 @@ which displays nicer looking diffs that the standard ``diff``
 tool. You can change that by passing ``--diff-command CMD`` to
 dune.
 
+Cram Tests
+==========
+
+Cram tests are expectation tests written in a shell like syntax. They are ideal
+for testing binaries. Cram tests are auto discovered from files or directories
+with a ``.t`` extension, so they must be enabled manuallly in the
+``dune-project`` file:
+
+.. code:: scheme
+
+   (lang dune 2.7)
+   (cram enable)
+
+
+File Tests
+----------
+
+To define a standalone test, we create a ``.t`` file. For example, ``foo.t``:
+
+.. code:: bash
+   Simplest possible cram test
+     $ echo "testing"
+
+This simple example demonstrates two components of cram tests:
+
+* Comments - Anything that doesn't start with a 2 space indentation is a comment
+
+* Commands - A command starts with 2 spaces followed by a ``$``. It is executed
+  in the shell and the output is diffed against the output below. In this
+  example, there's no output yet.
+
+To run the test and promote the results:
+
+.. code:: bash
+   $ dune runtest
```

Comment:
Using `dune` itself might be a bit meta for an introduction. Maybe something with `touch` and `ls`? or `echo` and `sort` so that there are no files?

Reply:
Just to clarify, I'm only telling the user the instructions to run the test here. I'm not telling the user to run dune as part of the test.

Reply:
Ah, I misread the block, I thought it was a cram test. All good then!